test/controllers/category_controller_test.exs

```diff
@@ -56,51 +46,67 @@ defmodule CodeCorps.CategoryControllerTest do
-      conn = post conn, category_path(conn, :create), build_payload(@valid_attrs)
-      assert json_response(conn, 401)
+      path = conn |> category_path(:create)
+      payload = json_payload(:category, @valid_attrs)
+
+      assert conn |> post(path, payload) |> json_response(401)
     end
 
     @tag :authenticated
     test "does not create resource and renders 401 when not authorized", %{conn: conn} do
-      conn = post conn, category_path(conn, :create), build_payload(@valid_attrs)
-      assert json_response(conn, 401)
+      path = conn |> category_path(:create)
+      payload = json_payload(:category, @invalid_attrs)
+      assert conn |> post(path, payload) |> json_response(401)
     end
   end
 
   describe "update" do
     @tag authenticated: :admin
     test "updates and renders chosen resource when data is valid", %{conn: conn} do
       category = insert(:category)
-      updated_attrs = %{@valid_attrs | description: "New Description"}
-      conn = put conn, category_path(conn, :update, category), build_payload(category.id, updated_attrs)
+      payload = json_payload(:category, @valid_attrs)
+      path = conn |> category_path(:update, category)
 
-      assert json_response(conn, 200)["data"]["id"] == "#{category.id}"
-      assert Repo.get_by(Category, updated_attrs)
+      response = conn |> put(path, payload) |> json_response(200)
+
+      assert response["data"]["id"] == "#{category.id}"
+      assert Repo.get_by(Category, @valid_attrs)
     end
 
     @tag authenticated: :admin
     test "does not update chosen resource and renders errors when data is invalid", %{conn: conn} do
       category = insert(:category)
-      conn = put conn, category_path(conn, :update, category), build_payload(category.id, @invalid_attrs)
-      assert json_response(conn, 422)["errors"] != %{}
+      payload = json_payload(:category, @invalid_attrs)
+      path = conn |> category_path(:update, category)
+
+      response = conn |> put(path, payload) |> json_response(422)
+
+      assert response["errors"] != %{}
     end
 
     test "does not update resource and renders 401 when not authenticated", %{conn: conn} do
       category = insert(:category)
-      conn = put conn, category_path(conn, :update, category), build_payload(category.id, @invalid_attrs)
-      assert json_response(conn, 401)
+      payload = json_payload(:category, @valid_attrs)
+      path = conn |> category_path(:update, category)
+
+      assert conn |> put(path, payload) |> json_response(401)
     end
 
     @tag :authenticated
     test "does not update resource and renders 401 when not authorized", %{conn: conn} do
       category = insert(:category)
-      conn = put conn, category_path(conn, :update, category), build_payload(category.id, @invalid_attrs)
-      assert json_response(conn, 401)
+      payload = json_payload(:category, @valid_attrs)
+      path = conn |> category_path(:update, category)
+
+      assert conn |> put(path, payload) |> json_response(401)
```

Comment:
Is there a better way to pipe the `conn` here that can be done in two lines instead of three?